client/controller.js

```diff
@@ -50,6 +55,8 @@ export function makeLoggedOutLayout( context, next ) {
  * divs.
  */
 export function clientRouter( route, ...middlewares ) {
+	routes.push( route );
+	routeMatcher = pathToRegexp( routes );
```

Comment:
>  unmounting as a side effect in state/ui/actions if the previous section was isomorphic, while the new one isn't?

That could be pretty tidy. Any thoughts on how/where we would do the re-render of the regular layout that is required after that unmount? This PR currently keeps it in client/boot/index.js to avoid the package split necessary when moving it into this file.